lib/rewards/server/plugin.js

```diff
@@ -0,0 +1,38 @@
+/**
+ *
+ * Reldens - Rewards Server Plugin
+ *
+ */
+const {PluginInterface} = require("../../features/plugin-interface");
+const {sc} = require("@reldens/utils");
+const {ObjectSubscriber} = require("./subscribers/object-subscriber");
+
+class RewardsPlugin extends PluginInterface
+{
+
+    setup(props)
+    {
+        this.events = sc.get(props, 'events', false);
+        if (!this.events) {
+            console.log('EventsManager undefined in RewardsPlugin.');
+        }
+
+        this.events.on('reldens.afterRunAdditionalRespawnSetup', async (eventArg) => {
+            await ObjectSubscriber.enrichWithRewards(eventArg.objInstance);
+        });
+
+        this.events.on('reldens.battleEnded', async (eventArg) => {
+            const {playerSchema, pve} = eventArg;
+            if (!pve.targetObject?.rewards) {
```

Comment:
create a variable:
```
let rewards = sc.get(pve.targetObject, 'rewards', false).
```
sc.get will validate pve.targetObject, then validate 'rewards', then if it doesn't exits then it will return 'false' as the default value

change the validation to scrict: `if(false === rewards)`